.github/workflows/ci.yml

```diff
@@ -76,12 +76,18 @@ jobs:
         pip install -r tests/requirements.txt
     - name: Test with pytest
       run: |
-        pytest --cov --junitxml=junit.xml
+        pytest --cov --junitxml=${{matrix.python-version}}junit.xml
     - name: Dogfooding codecov-cli
       if: ${{ !github.event.pull_request.head.repo.fork && github.repository_owner == 'codecov' }}
       run: |
         codecovcli -v do-upload --fail-on-error -t ${{ secrets.CODECOV_TOKEN }} --plugin pycoverage --flag python${{matrix.python-version}}
         codecovcli do-upload --report-type test_results --fail-on-error -t ${{ secrets.CODECOV_TOKEN }} --plugin pycoverage --flag python${{matrix.python-version}}
+    - name: Upload artifacts for test-results-processing
+      if: ${{ always() }}
```

Comment:
do you need this because the previous `run` command is optional?

Reply:
No. The reason for it is because the `Test with pytest` was failing the build and all subsequent steps were cancelled. Since the test processing is expecting failures, we need this step to run regardless of test failures.

Reply:
I think using `!cancelled()` may be better here because I think `always()` will make it so the step will run even if the job is cancelled